container/go/pkg/compat/overrideConfig.go

```diff
@@ -0,0 +1,530 @@
+	// User is the username to run the commands under.
+	User string
+	// Workdir is the working directory to set for the layer.
+	Workdir string
+	// NullEntrypoint indicates if there is an entrypoint.
+	// If true, Entrypoint will be set to null.
+	NullEntryPoint bool
+	// NullCmd indicates if there is a command.
+	// If true, Command will be set to null.
+	NullCmd bool
+	// OperatingSystem is the operating system to creater docker image for.
+	OperatingSystem string
+	// CreatedBy is the command that generated the image. Default bazel build...
+	CreatedBy string
+	// Author is the author of the image. Default bazel.
+	Author string
+	// LabelsArray is the labels used to augment those of the previous layer.
+	LabelsArray []string
+	// Ports are the ports used to augment those of the previous layer.
+	Ports []string
+	// Volumes are the volumes used to augment those of the previous layer.
+	Volumes []string
+	// EntrypointPrefix is the prefix to the entrypoint with the specified arguments.
+	EntrypointPrefix []string
+	// Env are the environments to augment those of the previous layer.
+	Env []string
+	// Command are the command(s) to override those of the previous layer.
+	Command []string
+	// Entrypoint are the new entrypoints to override entrypoint of previous layer.
+	Entrypoint []string
+	// Layer is the list of layer sha256 hashes that compose the image for which the config is written.
+	Layer []string
+	// StampInfoFile is a list of files from which to read substitutions of variables from.
+	StampInfoFile []string
+}
+
+// validate ensures that CreatedBy and Author are set.
+// based on current implementation, we expect them to never be empty.
+func (o *OverrideConfigOpts) validate() error {
+	if o.CreatedBy == "" {
+		return errors.New("CreatedBy was not specified")
+	}
+	if o.Author == "" {
+		return errors.New("Author was not specified")
+	}
+
+	return nil
+}
+
+// emptySHA256Digest returns the sha256 sum of an empty string.
+func emptySHA256Digest() (empty string) {
+	b := sha256.Sum256([]byte(""))
+	empty = hex.EncodeToString(b[:])
+	return
+}
+
+// extractValue returns the contents of a file pointed to by value if it starts with a '@'.
+func extractValue(value string) (string, error) {
+	if strings.HasPrefix(value, "@") {
+		f, err := os.Open(value[1:])
+		defer f.Close()
+		shaHash, err := ioutil.ReadAll(f)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to read content from file %s", value)
+		}
+		return string(shaHash), nil
+	}
+	return value, errors.Wrapf(nil, "unexpected value, got: %s want: @{...}", value)
+}
+
+// keyValueToMap converts an array of strings separated by '=' into a map of key-value pairs.
+func keyValueToMap(value []string) (map[string]string, error) {
+	convMap := make(map[string]string)
+	var temp []string
+	for _, kvpair := range value {
+		temp = strings.Split(kvpair, "=")
+		if len(temp) != 2 {
+			return convMap, errors.New("value in array are not of format key=value")
```

Comment:
include the vaue of kvpair in the error message for context. Something like `"%q is not in format key=value", kvpair`